src/flatten.ts

```diff
@@ -136,6 +137,22 @@ function stringify(this: ThisEncode, input: unknown, index: number) {
           } else if (input instanceof Promise) {
             str[index] = `["${TYPE_PROMISE}",${index}]`;
             deferred[index] = input;
+          } else if (input instanceof ReadableStream) {
+            str[index] = `["${TYPE_STREAM}",${index}]`;
+            // during this, the stream will be read, which is destructive
+            // so we tee it, and replace the methods on the original stream
+            // with methods forwarding to the left teed stream, while
+            // processing the right teed stream ourselves
+            const [left, right] = input.tee();
+            input.getReader = left.getReader.bind(left);
+            input.cancel = left.cancel.bind(left);
+            input.pipeThrough = left.pipeThrough.bind(left);
+            input.pipeTo = left.pipeTo.bind(left);
+            input.tee = left.tee.bind(left);
+            Object.defineProperty(input, "locked", {
+              get: () => left.locked,
+            });
+            streams[index] = right;
```

Comment:
This could also be omitted if there were a consensus that encoding is a destructive action - React does so - but I honestly felt a bit more comfortable with this `tee` approach.
```suggestion
            streams[index] = input;
```